lib/Dialects/LinalgTransform/Transforms/TransformInterpreter.cpp

```diff
@@ -546,26 +668,73 @@ findTransformTarget(OpTy transformOp, ModuleOp module,
   return applyPatternsOnce(module, std::move(patterns));
 }
 
+template <typename OpTy>
+static std::enable_if_t<!llvm::is_detected<has_target_matcher_t, OpTy>::value,
+                FailureOr<SmallVector<Operation *>>>
+matchUsingPDL(OpTy transformOp, ModuleOp module) {
+  assert(false && "matching with PDL on an unsupported op");
+}
+
+/// Returns the operations that `transformOp` should be run on. The
+/// `transformOp` is expected to have the TargetableTransformOpTrait. Its target
+/// may be either an SSA value that corresponds to a list of ops produced by
+/// other transforms and stored in `operations` or a list of ops that match the
+/// PDL pattern specified by its name.
+template <typename OpTy>
+static FailureOr<SmallVector<Operation *>>
+findTransformTarget(OpTy transformOp, ModuleOp module,
+                    const TransformOpMapping &operations) {
+  if (Value target = transformOp.target())
+    return operations.lookup(target);
+
+  return matchUsingPDL(transformOp, module);
+}
+
+template <typename ConfigOpTy>
+static void removeCurrentTarget(ConfigOpTy configOp,
+                                TransformOpMapping &operations) {
+  // Since there is only allowed use of the value in the transformation dialect,
+  // we can remove it from the mapping after processing its only user. This
+  // ensures we don't accidentally keep pointers to operations that may have
+  // been deleted by the current transformation.
+  if (!configOp.target())
+    return;
+  Value target = configOp.target();
+  assert(target.hasOneUse());
+  operations.erase(target);
+}
+
 /// Applies `transform` with options provided by `configOp` to all operations
 /// specified as targets by `configOp`.
-template <typename OpTy, typename FnTy>
-static LogicalResult
-executeTransformOnEach(ModuleOp module, OpTy configOp, FnTy transform,
-                       DenseMap<Value, SmallVector<LinalgOp>> &operations) {
-  FailureOr<SmallVector<LinalgOp>> targets =
+template <typename ConfigOpTy, typename FnTy,
+          typename TransformedOpType =
+              typename llvm::function_traits<FnTy>::template arg_t<0>,
+          typename ReturnType = typename llvm::function_traits<FnTy>::result_t>
+static std::enable_if_t<!std::is_same<ReturnType, LogicalResult>::value,
+                        LogicalResult>
+executeTransformOnEach(ModuleOp module, ConfigOpTy configOp, FnTy transform,
```

Comment:
What is the value of the sfinae genuflexion here?
Can we have 2 functions with different names instead ?